File: tests/fixtures/check_then_act/ts/stale_read_checked_late_fixture.ts
// The read happens before the await and the check after it: the check tests stale data.
export async function settle(repo: any, id: string) {
  const order = repo.get(id);
  await repo.sync(id); // expect: gap await
  if (order.open) { // expect: check
    repo.update(id, { open: false }); // expect: write
  }
}

File: tests/fixtures/check_then_act/py/stale_read_checked_late_fixture.py
# The read happens before the await and the check after it: the check tests stale data.
async def settle(repo, order_id):
    order = repo.get(order_id)
    await repo.sync(order_id)  # expect: gap await
    if order.open:  # expect: check
        repo.update(order_id, open=False)  # expect: write
